classes/controller/FrontController.php

```diff
@@ -1622,6 +1622,26 @@ public function getTemplateVarCustomer($customer = null)
         return $cust;
     }
 
+    public function getShopLogo(): array
+    {
+        $logoData = [];
+        $urls = $this->getTemplateVarUrls();
+        $psImageUrl = $urls['img_ps_url'] ?? _PS_IMG_;
+        $logoDir = _PS_IMG_DIR_;
+
+        if (Configuration::hasKey('PS_LOGO')) {
+            $logoFileName = Configuration::get('PS_LOGO');
+            $logoFileDir = $logoDir . $logoFileName;
+            $logoSizes = getimagesize($logoFileDir);
+
+            $logoData['src'] = $psImageUrl . $logoFileName;
+            $logoData['width'] = $logoSizes[0];
+            $logoData['height'] = $logoSizes[1];
+        }
+
+        return $logoData;
+    }
```

Comment:
```suggestion
    /**
     * Get the shop logo with its dimensions
     *
     * @return array
     */
    public function getShopLogo(): array
    {
        if (!Configuration::hasKey('PS_LOGO')) {
            return [];
        }
        
        $logoFileName = Configuration::get('PS_LOGO');
        list($logoWidth, $logoHeight) = getimagesize(_PS_IMG_DIR_ . $logoFileName);
        
        return [
            'src' => ($this->getTemplateVarUrls()['img_ps_url'] ?? _PS_IMG_) . $logoFileName,
            'width' => $logoWidth,
            'height' => $logoHeight
        ];
    }
```

@Oksydan WDYT? :D

Reply:
@kpodemski You just made me feel like I know nothing about PHP 😂

Reply:
@Oksydan no, no, it's just a little optimization, you don't need to set variables if you use them once, return early if no need to continue, etc. 👍